pkg/config/config_test.go

```diff
@@ -1,63 +1,128 @@
 package config
 
 import (
-	"os"
 	"testing"
+
+	"github.com/stretchr/testify/require"
 )
 
 func TestLoadConfig(t *testing.T) {
-	// Set up a temporary config file
-	configContent := `
-server_groups:
-  - name: loki1
-    url: http://loki1.example.com
-    timeout: 10
-    headers:
-      Authorization: Bearer token1
-  - name: loki2
-    url: http://loki2.example.com
-    timeout: 15
-    headers:
-      Authorization: Bearer token2
-logging:
-  level: info
-  format: json
-`
-	configFile, err := os.CreateTemp("", "config.yaml")
-	if err != nil {
-		t.Fatalf("Failed to create temp config file: %v", err)
-	}
-	defer os.Remove(configFile.Name())
+	tests := []struct {
+		name         string
+		configFile   string
+		wantErr      bool
+		validateFunc func(*testing.T, *Config)
+	}{
+		{
+			name:       "valid basic config",
+			configFile: "testdata/valid_config.yaml",
+			wantErr:    false,
+			validateFunc: func(t *testing.T, cfg *Config) {
+				require.NotNil(t, cfg)
```

Comment:
I thing we can remove this require.NotNil(t, cfg) since is already on the range as in https://github.com/paulojmdias/lokxy/blob/398058da8d01489f8ed7a312a3d7ed43aaff23f9/pkg/config/config_test.go#L121

```go
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			cfg, err := LoadConfig(tt.configFile)

(...)
			require.NotNil(t, cfg)
```